### PR TITLE
Add TypeNameHelper to pretty print type names

### PR DESCRIPTION
**Pull request by smitpatel**
@bricelam

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@smitpatel__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Smit Patel). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla2.msopentech.com.
    </p>

TTYL, MSOTBOT;

**Comment by bricelam** [comment]
:shipit:

File: src/Microsoft.Framework.TypeNameHelper.Internal/TypeNameHelper.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        public static string GetTypeDisplayName(Type type, bool fullname = true)
+        {
+            var sb = new StringBuilder();
+            ProcessTypeName(type, sb, fullname);
+            return sb.ToString();
+        }
+
+        private static void AppendGenericArguments(Type[] args, int startIndex, int nArgsToAppend, StringBuilder sb, bool fullName)
+        {
+            var nArgs = args.Length;
```
Comment (Eilon) [inline src/Microsoft.Framework.TypeNameHelper.Internal/TypeNameHelper.cs]: Can we switch all this away from using Hungarian notation? Not sure how that got here.

Comment (smitpatel) [inline src/Microsoft.Framework.TypeNameHelper.Internal/TypeNameHelper.cs]: This is already merged. Should I make corrections and submit new pull req for it?

Comment (Eilon) [inline src/Microsoft.Framework.TypeNameHelper.Internal/TypeNameHelper.cs]: Yeah that's fine.

Comment (smitpatel) [inline src/Microsoft.Framework.TypeNameHelper.Internal/TypeNameHelper.cs]: created #16

Comment (Eilon) [inline src/Microsoft.Framework.TypeNameHelper.Internal/TypeNameHelper.cs]: Thanks, signed off on that one.

**Pull request merged**